crates/bevy_pbr/src/atmosphere/functions.wgsl

```diff
@@ -334,9 +349,16 @@ fn ndc_to_uv(ndc: vec2<f32>) -> vec2<f32> {
 }
 
 /// Converts a direction in world space to atmosphere space
-fn direction_world_to_atmosphere(dir_ws: vec3<f32>) -> vec3<f32> {
-    let dir_as = atmosphere_transforms.atmosphere_from_world * vec4(dir_ws, 0.0);
-    return dir_as.xyz;
+fn direction_world_to_atmosphere(dir_ws: vec3<f32>, up: vec3<f32>) -> vec3<f32> {
+    // Camera forward in world space (-Z in view to world transform)
+    let forward_ws = normalize((view.world_from_view * vec4(0.0, 0.0, -1.0, 0.0)).xyz);
+    let tangent_z = up * dot(forward_ws, up) - forward_ws;
```

Comment:
ah, the normalize here should be on this line not the former. world_from_view z should be unit, but the vector rejection of forward vs up isnt. I'd normalize tangent_z but not forward_ws

Reply:
Changed this to the suggested version